coffee_cmd/src/coffee/mod.rs

```diff
@@ -180,19 +180,27 @@ impl PluginManager for CoffeeManager {
     }
 
     async fn list(&mut self, remotes: bool) -> Result<Value, CoffeeError> {
-        let mut plugin_names: Vec<Plugin> = Vec::new();
+        let installed_plugins_vec: Vec<Plugin> = self.config.plugins.clone();
+        let mut plugin_json = json!({
+            "plugins": serde_json::to_value(&installed_plugins_vec).unwrap(),
+         });
         if remotes {
+            let mut remote_list: Vec<Value> = Vec::new();
             for repo in &self.repos {
-                let mut repolist = repo.list().await.unwrap();
-                plugin_names.append(&mut repolist);
+                let remote_repo_json = json!(
+                {
+                    "local_name": repo.get_name().await.unwrap(),
+                    "link": repo.get_url().await.unwrap().url_string,
+                    "plugins": repo.list().await.unwrap(),
+                 });
+                remote_list.push(remote_repo_json);
             }
-        } else {
-            plugin_names = self.config.plugins.clone();
+            plugin_json = json!({
+                "installed plugins": serde_json::to_value(&installed_plugins_vec).unwrap(),
```

Comment:
```suggestion
                "plugins": serde_json::to_value(&installed_plugins_vec).unwrap(),
```

space as key are illegal when you are returning json